typing/typedecl.ml

```diff
@@ -78,10 +78,10 @@ let get_unboxed_from_attributes sdecl =
   let boxed = Builtin_attributes.has_boxed sdecl.ptype_attributes in
   match boxed, unboxed, !Clflags.unboxed_types with
   | true, true, _ -> raise (Error(sdecl.ptype_loc, Boxed_and_unboxed))
-  | true, false, _ -> unboxed_false_default_false
-  | false, true, _ -> unboxed_true_default_false
-  | false, false, false -> unboxed_false_default_true
-  | false, false, true -> unboxed_true_default_true
+  | true, false, _ -> false, false
+  | false, true, _ -> true, false
+  | false, false, false -> false, true
+  | false, false, true -> true, true
```

Comment:
It's unclear what this function does, a comment could help (or a record type just for this function?). I also believe that the comments lines 73-74 is outdated and should be removed.